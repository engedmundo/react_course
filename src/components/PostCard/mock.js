export const postCardPropsMock = {
    title: "Some title",
    body: "Some body",
    id: 1,
    cover: "img/img.jpg"
};