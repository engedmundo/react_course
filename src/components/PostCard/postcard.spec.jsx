import { render, screen } from '@testing-library/react';
import { Postcard } from '.';
import { postCardPropsMock } from './mock';
// import userEvent from '@testing-library/user-event';

const props = postCardPropsMock;


describe("<Postcard />", () => {
  test("render info successfully", () => {
    // const { debug } = render(<Postcard { ...props }/>);
    // debug();
    // expect.assertions(1);
    
    render(<Postcard { ...props }/>);
    
    const img = screen.getByRole("img", { name: "Some title"});
    expect(img).toBeInTheDocument();
    expect(img).toHaveAttribute("src", "img/img.jpg");

    const heading = screen.getByRole("heading", { name: "Some title"});
    expect(heading).toBeInTheDocument();

    const paragraph = screen.getByText("Some body");
    expect(paragraph).toBeInTheDocument();
  });
  
  test("to match snapshot", () => {
    const { container } = render (<Postcard { ...props } />);
    expect(container.firstChild).toMatchSnapshot();
  });
});