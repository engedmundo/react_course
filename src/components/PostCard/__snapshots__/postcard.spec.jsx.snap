// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Postcard /> to match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="Some title"
    src="img/img.jpg"
  />
  <div
    class="post-content"
  >
    <h1>
      Some title
    </h1>
    <p>
      Some body
    </p>
  </div>
</div>
`;
